### PR TITLE
Erasure fixes and add erasure to ci

### LINKED ISSUE #300: Add erasure coding
[ ] Get erasure coding working
[ ] CI for `--features=erasure`

### PR DESCRIPTION
**Pull request by sakridge**
Fixes #300

### CODE REVIEW

**Comment by garious** [comment]
Thanks @mvines for the the bash checker. Already paying dividends!

File: src/crdt.rs [hunk 1]
```diff
@@ -164,12 +164,41 @@ impl Crdt {
...
+        transmit_index: &mut u64,
+    ) -> Result<()> {
+        let me: ReplicatedData = {
+            let robj = obj.read().expect("'obj' read lock in crdt::index_blobs");
+            info!("broadcast table {}", robj.table.len());
+            robj.table[&robj.me].clone()
+        };
+
+        // enumerate all the blobs, those are the indices
+        let orders: Vec<_> = blobs.iter().enumerate().collect();
```
Comment (garious) [inline src/crdt.rs]: No need for that `collect()` and its associated malloc.

File: src/crdt.rs [hunk 2]
```diff
@@ -164,12 +164,41 @@ impl Crdt {
...
+        info!("orders table {}", orders.len());
+        let _: Vec<_> = orders
+            .into_iter()
+            .map(|(i, b)| {
```
Comment (garious) [inline src/crdt.rs]: Use `for_each` or a for loop when you don't need a new list.

File: src/streamer.rs [hunk 3]
```diff
@@ -394,6 +397,18 @@ fn broadcast(
             win[pos] = Some(b);
         }
     }
+
+    // Fill in the coding blob data from the window data blobs
+    #[cfg(feature = "erasure")]
```
Comment (garious) [inline src/streamer.rs]: Can you add this to the list of `--features` anywhere that we currently mention `cuda`?

File: src/streamer.rs [hunk 4]
```diff
@@ -394,6 +397,18 @@ fn broadcast(
...
+    {
+        match erasure::generate_coding(&mut window.write().unwrap(), *transmit_index as usize) {
```
Comment (garious) [inline src/streamer.rs]: `is_err()`

**Comment by sakridge** [comment]
Want to check that i didn't break the broadcast stuff.

**Pull request merged**